api/serializers.py

```diff
@@ -367,41 +213,42 @@ class Meta:
 
     # this creates a record from the json, modifying the keys
     def create(self, validated_data):
-        validated_data['status'] = validated_data.pop('status_id', None)
-        validated_data['assigned_user'] = validated_data.pop('assigned_user_id', None)
-        validated_data['team'] = validated_data.pop('team_id')
-
-        # this will remove the entry even if the call requesting using the
-        # serializer does not use the tag_id_list
-        validated_data.pop('tag_id_list', None)
-        validated_data.pop('parent_id', None)
+        tag_list = validated_data.pop('tag_list', None)
 
         ticket = api_models.Ticket.objects.create(
             **validated_data
         )
 
+        api_models.TicketTag.create_all(tag_list, ticket)
+
         return ticket
 
     # update the instance with validated_data
     def update(self, instance, validated_data):
-        validated_data['status'] = validated_data.pop('status_id', None)
-        validated_data['assigned_user'] = validated_data.pop('assigned_user_id', None)
-        validated_data.pop('tag_id_list', None)
+        tag_list = validated_data.pop('tag_list', None)
+
+        api_models.TicketTag.delete_difference(tag_list, instance)
+
         return super().update(instance, validated_data)
 
 
 class EventSerializer(serializers.ModelSerializer):
+    """
+    On writes,\n
+    - user expects a primary key\n
+
+    On reads,\n
+    - user is a serialized user object\n
+
+    The autogenerated documentation does not account for this
+    """
+
     id = serializers.ReadOnlyField()
-    team_id = serializers.PrimaryKeyRelatedField(
-        many=False, read_only=False, queryset=api_models.Team.objects.all()
-    )
     created = serializers.ReadOnlyField()
     event_type = serializers.ReadOnlyField()
-    user = UserSerializer(many=False, read_only=True)
-    user_id = serializers.PrimaryKeyRelatedField(
-        many=False, write_only=True, required=False, queryset=get_user_model().objects.all()
+    user = PrimaryKeySerializedField(
+        many=False, queryset=api_models.Member.objects.all(), serializer=UserSerializer
```

Comment:
read_only=True seems to still be in Mason's branch and not in this PR.

Reply:
updating. events aren't a huge feature for the slack bot and probably won't be integrated until isaac starts with some of his changes.